### PR TITLE
[DEV-3352] replace bankAccounts with bankData

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations (on [DFX services](https://dev.services.dfx.swiss))
  - Login/logout
  - Buy/sell payment request
  - KYC page

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Comment by Yannick1712** [comment]
- [ ] add migration

File: src/subdomains/generic/user/models/bank-data/bank-data.service.ts [hunk 1]
```diff
@@ -243,6 +245,9 @@ export class BankDataService {
     const multiIbans = await this.specialAccountService.getMultiAccountIbans();
     if (multiIbans.includes(dto.iban)) throw new BadRequestException('Multi-account IBANs not allowed');
 
+    if (!(await this.isValidIbanCountry(dto.iban)))
+      throw new BadRequestException('Iban country is currently not supported');
+
```
Comment (davidleomay) [inline src/subdomains/generic/user/models/bank-data/bank-data.service.ts:250]: `IBAN country ...`

File: src/subdomains/core/buy-crypto/routes/buy/buy.service.ts [hunk 2]
```diff
@@ -30,6 +34,26 @@ export class BuyService {
     await this.buyRepo.update({ annualVolume: Not(0) }, { annualVolume: 0 });
   }
 
+  @Cron(CronExpression.EVERY_MINUTE)
+  @Lock(1800)
+  async checkCryptoPayIn() {
+    if (DisabledProcess(Process.BUY_BANK_ACCOUNT_SYNC)) return;
```
Comment (davidleomay) [inline src/subdomains/core/buy-crypto/routes/buy/buy.service.ts:40]: Copy&paste 😉

File: src/subdomains/core/sell-crypto/route/sell.service.ts [hunk 3]
```diff
@@ -48,8 +51,29 @@ export class SellService {
     @Inject(forwardRef(() => TransactionUtilService))
     private readonly transactionUtilService: TransactionUtilService,
     private readonly routeService: RouteService,
+    private readonly bankDataService: BankDataService,
   ) {}
 
+  @Cron(CronExpression.EVERY_MINUTE)
+  @Lock(1800)
+  async checkCryptoPayIn() {
+    if (DisabledProcess(Process.SELL_BANK_ACCOUNT_SYNC)) return;
```
Comment (davidleomay) [inline src/subdomains/core/sell-crypto/route/sell.service.ts:60]: Copy&paste 😉

**Review by davidleomay — APPROVED** [review]

**Review by davidleomay — APPROVED** [review]

**Pull request merged**